services/api/src/messaging/players.service.ts

```diff
@@ -108,6 +113,10 @@ const moveSkipped: Handler = (ws, { userId }: MoveSkippedMessage): void => {
   try {
     const [playerConnection, player] = findPlayerById(ws.roomId, userId);
 
+    if (player.role === 'Spectator') {
```

Comment:
instead of this, we should come up with a mechanisim to block a list of actions for specific role.